### PR TITLE
BridgedNetwork implementation

### PR DESCRIPTION
**Pull request by Raikerian**
## Which issue(s) this PR fixes:

Found one issue related to that https://github.com/Code-Hex/vz/issues/138

## Additional documentation

Added implementation for BridgedNetwork in order to be able to create network device using bridge network, rather than NAT.
Using BridgedNetwork does indeed require special entitlements from Apple, which is a whole PIA to setup and test by itself, as you need proper certificates and signing in place after you got it added to your Organization by Apple. My current project does have those entitlements and needs bridged network, so I was able to test this and it does the job. Please let me know your thoughts if anything needs to be improved.

### CODE REVIEW

File: virtualization_11.m [hunk 1]
```diff
@@ -392,6 +392,50 @@ void setStorageDevicesVZVirtualMachineConfiguration(void *config,
...
+}
+
+/*!
+ @abstract Return a display name if available (e.g. "Ethernet").
+ */
+void *VZBridgedNetworkInterface_localizedDisplayName(void *networkInterface)
+{
+    if (@available(macOS 11, *)) {
+        return [(VZBridgedNetworkInterface *)networkInterface localizedDisplayName]; // NSString *
+    }
```
Comment (Code-Hex) [inline virtualization_11.m:434]: Please fix to return `const char *`

File: virtualization_11.m [hunk 2]
```diff
@@ -392,6 +392,50 @@ void setStorageDevicesVZVirtualMachineConfiguration(void *config,
...
+}
+
+/*!
+ @abstract Return the unique identifier for this interface. The identifier is the BSD name associated with the interface (e.g. "en0").
+ */
+void *VZBridgedNetworkInterface_identifier(void *networkInterface)
+{
+    if (@available(macOS 11, *)) {
+        return [(VZBridgedNetworkInterface *)networkInterface identifier]; // NSString *
+    }
```
Comment (Code-Hex) [inline virtualization_11.m:422]: Please fix to return `const char *`

File: internal/objc/objc.go [hunk 3]
```diff
@@ -142,6 +152,27 @@ func (n *NSArray) ToPointerSlice() []unsafe.Pointer {
...
+// CString converts *C.char from *NSString
+func (s *NSString) CString() *C.char {
+	return C.CStringFromNSString(s.ptr())
+}
+
+// String converts Go string from *NSString
+func (s *NSString) String() string {
+	length := C.LengthFromNSString(s.ptr())
+	return C.GoStringN(s.CString(), C.int(length))
+}
```
Comment (Code-Hex) [inline internal/objc/objc.go:174]: These function definitions are not necessary.

File: network.go [hunk 4]
```diff
@@ -39,6 +39,42 @@ type BridgedNetwork interface {
...
+}
+
+func (b *baseBridgedNetwork) Identifier() string {
+	nsStirng := objc.NewNSString(
+		C.VZBridgedNetworkInterface_identifier(objc.Ptr(b)),
+	)
+	return nsStirng.String()
+}
+
+func (b *baseBridgedNetwork) LocalizedDisplayName() string {
```
Comment (Code-Hex) [inline network.go:71]: Could you add a GoDoc?

**Review by Code-Hex — CHANGES_REQUESTED** [review]

**Comment by Raikerian** [comment]
@Code-Hex apologies for the delay, all the comments are now addressed

**Review by Code-Hex — APPROVED** [review]
Thanks

**Comment by Code-Hex** [comment]
I will release tomorrow in JST

**Pull request merged**